Source/WebKit/UIProcess/WebFrameProxy.cpp

```diff
@@ -400,7 +400,8 @@ void WebFrameProxy::commitProvisionalFrame(FrameIdentifier frameID, FrameInfoDat
     m_provisionalFrame->process().provisionalFrameCommitted(*this);
     send(Messages::WebFrame::DidCommitLoadInAnotherProcess());
     m_process->removeMessageReceiver(Messages::WebFrameProxy::messageReceiverName(), m_frameID.object());
-    m_parentFrameProcess = std::exchange(m_process, m_provisionalFrame->process());
+    m_parentFrameProcess = m_process.ptr();
```

Comment:
Either the naming is wrong or the logic is wrong. Currently, this is not the parent frame process but the process before the last swap.

Sure, on the first swap, this happens to be the parent frame process. However, on later swap, this is no longer true.

Reply:
Naming is probably not ideal, but this is the process which hosts the parent frame if it is different than this frame.  It is all likely to be considerably reworked in the coming weeks.

Reply:
But this is not true, it is the process that was used by the frame before the *last* process swap. It is not necessarily the parent frame process. I think you should initialize m_parentFrameProcess differently.

Presumably, you could just initialize m_parentFrameProcess on WebFrameProcess construction to be the same as m_process for example. And then no worry about updating it later?

Reply:
Or not overwrite `m_parentFrameProcess` if it is already set. If you don't want to eagerly initialize m_parentFrameProcess in the constructor for some reason.

Reply:
I'm going to remove this change from this PR and defer this conversation.  This PR is about the test.